legal-api/src/legal_api/resources/v2/namerequest.py

```diff
@@ -40,3 +41,40 @@ def get(identifier):
         current_app.logger.error(err)
         abort(500)
         return {}, 500  # to appease the linter
+
+
+@bp.route('/<string:identifier>/validate', methods=['GET'])
+@cross_origin(origin='*')
+def validate_with_contact_info(identifier):
+    """Return a JSON object with name request information."""
+    try:
+        nr_response = namex.query_nr_number(identifier)
+        # Errors in general will just pass though,
+        # 404 is overriden as it is giving namex-api specific messaging
+        if nr_response.status_code == 404:
+            return make_response(jsonify(message='{} not found.'.format(identifier)), 404)
+
+        nr_json = nr_response.json()
+
+        # Check the NR is affiliated with this account
+        orgs_response = AccountService.get_account_by_affiliated_identifier(identifier)
+        if len(orgs_response['orgs']):
+            return jsonify(nr_json)
+
+        # If NR is not affiliated, validate the email and phone
+        email = request.args.get('email', None)
+        phone = request.args.get('phone', None)
+        if not (email or phone):
+            # The request must include email or phone number
+            return make_response(jsonify(message='The request must include email or phone number.'), 400)
```

Comment:
I think you should check this before line 51, otherwise we are giving the caller some information (ie, NR doesn't exist) before checking validation.

Reply:
Updated